### PR TITLE
Adding generated Compute json schema based on 2016-04-30-preview spec

### PR DESCRIPTION
**Pull request by waynekuo**
- Adding generated Compute json schema based on 2016-04-30-preview spec
- Adding 2016-04-30-preview unit tests for
  - VM
  - AvailabilitySet
  - Managed DataDisk
  - Snapshots
  - Images
  - VM and VMSS with Managed data disks
- Adding 2016-04-30-preview resource types to deploymentTemplate.json

### CODE REVIEW

**Comment by hyonholee** [comment]
LGTM.  Thanks.

File: schemas/2016-04-30-preview/Microsoft.Compute.json [hunk 1]
```diff
@@ -0,0 +1,2760 @@
...
+              "$ref": "#/definitions/Sku"
+            },
+            {
+              "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+            }
+          ],
+          "description": "Sku of the availability set"
+        }
+      },
+      "required": [
```
Comment (TianoMS) [inline schemas/2016-04-30-preview/Microsoft.Compute.json]: >required [](start = 7, length = 8)

Isn't "location" a required property?

File: schemas/2016-04-30-preview/Microsoft.Compute.json [hunk 2]
```diff
@@ -0,0 +1,2760 @@
...
+            {
+              "$ref": "#/definitions/VirtualMachineScaleSetProperties"
+            },
+            {
+              "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+            }
+          ]
+        }
+      },
+      "required": [
```
Comment (TianoMS) [inline schemas/2016-04-30-preview/Microsoft.Compute.json]: >required [](start = 7, length = 8)

Is "sku" and "plan" required?

File: schemas/2016-04-30-preview/Microsoft.Compute.json [hunk 3]
```diff
@@ -0,0 +1,4781 @@
...
+        "id": {
+          "type": "string",
+          "description": "Resource Id"
+        }
+      }
+    },
+    "UpgradePolicy": {
+      "type": "object",
+      "properties": {
+        "mode": {
```
Comment (TianoMS) [inline schemas/2016-04-30-preview/Microsoft.Compute.json]: "mode" is a required property in the previous version, and can be any string value. Can we confirm on the change?

File: schemas/2016-04-30-preview/Microsoft.Compute.json [hunk 4]
```diff
@@ -0,0 +1,4781 @@
...
+        },
+        "name": {
+          "type": "string",
+          "description": "The IP configuration name."
+        },
+        "properties": {
+          "oneOf": [
+            {
+              "$ref": "#/definitions/VirtualMachineScaleSetIPConfigurationProperties"
+
```
Comment (TianoMS) [inline schemas/2016-04-30-preview/Microsoft.Compute.json]: "upgradePolicy" and "virtualMachineProfile" are required properties in the previous. Can we confirm on the change?

File: schemas/2016-04-30-preview/Microsoft.Compute.json [hunk 5]
```diff
@@ -0,0 +1,4781 @@
...
+          ],
+          "description": "The virtual machine extension instance view."
+        }
+      },
+      "description": "Describes the properties of a Virtual Machine Extension."
+    },
+    "VirtualMachineProperties": {
+      "type": "object",
+      "properties": {
+        "hardwareProfile": {
```
Comment (TianoMS) [inline schemas/2016-04-30-preview/Microsoft.Compute.json]: "hardwareProfile", "storageProfile" and "networkProfile" are required properties in the previous. Can we confirm on the change?

File: schemas/2016-04-30-preview/Microsoft.Compute.json [hunk 6]
```diff
@@ -0,0 +1,4781 @@
...
+            { "$ref": "#/definitions/site24x7ApmInsightExtn"},
+            { "$ref": "#/definitions/trendMicroDSALinux"},
+            { "$ref": "#/definitions/trendMicroDSA"},
+            { "$ref": "#/definitions/bmcCtmAgentLinux"},
+            { "$ref": "#/definitions/bmcCtmAgentWindows"},
+            { "$ref": "#/definitions/OSPatchingForLinux"},
+            { "$ref": "#/definitions/VMSnapshot"},
+            { "$ref": "#/definitions/VMSnapshotLinux"},
+            { "$ref": "#/definitions/customScript" }
+          ]
```
Comment (TianoMS) [inline schemas/2016-04-30-preview/Microsoft.Compute.json]: Justin recently added two new extension types. Seems those two are missing here. Please sync the 2015-08-01 version schema and move from there.

**Comment by TianoMS** [comment]
Left a few comments. Most are related to the difference with previous version. We need to make sure whether the new version is accurate.

File: schemas/2016-04-30-preview/Microsoft.Compute.json (hunk 6 above)
Comment (jluk) [inline schemas/2016-04-30-preview/Microsoft.Compute.json]: Yes please add and the schemas related for them (as well as tests):
            { "$ref": "#/definitions/networkWatcherAgentWindows"}
            { "$ref": "#/definitions/networkWatcherAgentLinux"}

Comment (waynekuo) [inline schemas/2016-04-30-preview/Microsoft.Compute.json]: Done

File: schemas/2016-04-30-preview/Microsoft.Compute.json (hunk 2 above)
Comment (waynekuo) [inline schemas/2016-04-30-preview/Microsoft.Compute.json]: Only sku is required

File: schemas/2016-04-30-preview/Microsoft.Compute.json (hunk 1 above)
Comment (waynekuo) [inline schemas/2016-04-30-preview/Microsoft.Compute.json]: Done

File: schemas/2016-04-30-preview/Microsoft.Compute.json (hunk 3 above)
Comment (waynekuo) [inline schemas/2016-04-30-preview/Microsoft.Compute.json]: Let's check with the VMSS/CRP dev team.

File: schemas/2016-04-30-preview/Microsoft.Compute.json (hunk 5 above)
Comment (waynekuo) [inline schemas/2016-04-30-preview/Microsoft.Compute.json]: Created another thread for this with CRP devs.

File: schemas/2016-04-30-preview/Microsoft.Compute.json (hunk 4 above)
Comment (waynekuo) [inline schemas/2016-04-30-preview/Microsoft.Compute.json]: Checking with VMSS team.

File: schemas/2016-04-30-preview/Microsoft.Compute.json [hunk 7]
```diff
@@ -0,0 +1,4781 @@
...
+        }
+      },
+      "required": [
+        "type",
+        "properties",
+        "apiVersion"
```
Comment (waynekuo) [inline schemas/2016-04-30-preview/Microsoft.Compute.json]: Unclear if Extensions require 'location'. They were not required previously.

File: schemas/2016-04-30-preview/Microsoft.Compute.json [hunk 8]
```diff
@@ -0,0 +1,4781 @@
...
+            { "$ref": "#/definitions/datadogWindowsAgent"},
+            { "$ref": "#/definitions/dockerExtension"},
+            { "$ref": "#/definitions/customScript" }
+          ]
+        }
+      },
+      "required": [
+        "type",
+        "apiVersion",
+        "properties"
```
Comment (waynekuo) [inline schemas/2016-04-30-preview/Microsoft.Compute.json]: Unclear if Extensions require 'location'. They were not required previously.

File: schemas/2016-04-30-preview/Microsoft.Compute.json (hunk 7 above)
Comment (jluk) [inline schemas/2016-04-30-preview/Microsoft.Compute.json]: CRP (maybe Nathan) would be the right person to answer this question

Comment (TianoMS) [inline schemas/2016-04-30-preview/Microsoft.Compute.json]: There's a recent change which adds location in the required list. You can sync master and get that.

Comment (waynekuo) [inline schemas/2016-04-30-preview/Microsoft.Compute.json]: Done.

File: schemas/2016-04-30-preview/Microsoft.Compute.json (hunk 3 above)
Comment (waynekuo) [inline schemas/2016-04-30-preview/Microsoft.Compute.json]: Not needed.

File: schemas/2016-04-30-preview/Microsoft.Compute.json (hunk 5 above)
Comment (waynekuo) [inline schemas/2016-04-30-preview/Microsoft.Compute.json]: Not needed.

File: schemas/2016-04-30-preview/Microsoft.Compute.json [hunk 9]
```diff
@@ -0,0 +1,4843 @@
...
+              "$ref": "#/definitions/SshConfiguration"
+            },
+            {
+              "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+            }
+          ],
+          "description": "The SSH configuration for linux VMs."
+        }
+      },
+      "description": "Describes Windows configuration of the OS Profile."
```
Comment (ajithps) [inline schemas/2016-04-30-preview/Microsoft.Compute.json]: Shouldn't this be `Describes Linux configuration`?

**Comment by TianoMS** [comment]
>               "$ref": "#/definitions/DiskProperties"

Is it correct to use DiskProperties for snapshots?

---
Refers to: schemas/2016-04-30-preview/Microsoft.Compute.json:373 in 8d57617. [](commit_id = 8d576174ecb68a5a88fd5b0e48b86411c36d76c2, deletion_comment = False)

**Comment by TianoMS** [comment]
@waynekuo Please address the remaining comments. I'll merge this PR today.

File: schemas/2016-04-30-preview/Microsoft.Compute.json (hunk 9 above)
Comment (waynekuo) [inline schemas/2016-04-30-preview/Microsoft.Compute.json]: This is a bug in the Swagger spec. Fixed it on both sides.

**Comment by waynekuo** [comment]
This is correct.

**Pull request merged**